python/cuml/cuml/fil/fil.pyx

```diff
@@ -1224,6 +1244,9 @@ class ForestInference(UniversalBase, CMajorInputTagMixin):
                 result = GlobalSettings().xpy.argmax(
                     proba.to_output(output_type='array'), axis=1
                 )
+            if len(result.shape) == 2 and result.shape[-1] == 1:
+                result = result.flatten()
```

Comment:
Why are we flattening here?

Reply:
Without this line, we'd break old tests in `test_fil_compat.py`. This line was already in `experimental/fil/fil.pyx` before the migration.

Reply:
Ah you're right, then let's revise the tests in `test_fil_compat.py`